includes/class-wc-api.php

```diff
@@ -23,6 +23,9 @@ class WC_API extends WC_Legacy_API {
 	public function __construct() {
 		parent::__construct();
 
+		// Wrangle long requests.
+		add_action( 'shutdown', array( $this, 'rest_send_reponse_before_shutdown' ), -1 );
```

Comment:
If the slowdown is coming from Jetpack then yes, certainly we want to fix it there.